Add password reset flow

fjvallarino opened the pull request:
@ezequielc @alex-deaconu this PR does not have the unit tests yet, but I wanted to submit it now so Alex can provide its input.

The console.* calls will be replaced with $log.*. The ctr-confirm-account error handling is the most conflicting part at this point. We need some way of sending an error code to the login screen (since we don't have an intermediate page for displaying errors in case they happen).

alex-deaconu commented on src/userstate/app.js:
```diff
@@ -86,6 +86,39 @@
+        template: "<div ui-view></div>"
+      })
+
+      .state("common.auth.requestpasswordreset.final", {
```
@fjvallarino I don't think we need a final state here, as the `state` variable is not used in the controller. Thanks.

alex-deaconu commented on src/userstate/controllers/ctr-request-password-reset.js:
```diff
@@ -0,0 +1,29 @@
+angular.module("risevision.common.components.userstate")
+  .controller("RequestPasswordResetCtrl", ["$scope", "$loading", "$stateParams",
+    "userauth",
+    function ($scope, $loading, $stateParams, userauth) {
```
@fjvallarino `$stateParams` is not used in this controller; I think we can remove. Thanks.

fjvallarino commented:
@alex-deaconu I applied the changes you proposed, and pushed unit testing for my changes.

I don't really understand why coverage decreased by 64%, since I added tests for all the code I added...

alex-deaconu commented on test/unit/userstate/controllers/ctr-reset-password-confirm.tests.js:
```diff
@@ -0,0 +1,161 @@
+"use strict";
+describe.only("controller: Reset Password Confirm", function() {
```
@fjvallarino this may be the cause for the loss of test coverage. Thanks.

fjvallarino replied:
@alex-deaconu that was the issue. Thanks!

ezequielc commented:
@fjvallarino Is this ready for review? Do you have a staged version? Thanks

ezequielc commented:
@fjvallarino Code looks good. Let me about staged version. Thanks!

fjvallarino commented:
@ezequielc sorry. The staged version is at: https://apps-stage-3.risevision.com

ezequielc commented:
@fjvallarino I think the staged version is pointing to a non-existent Core version (https://registration-reset-emails-dot-rvacore-test.appspot.com). I saw there is a 'registration-reset-emails-2' version. Do you mind updating it? Thanks!

fjvallarino commented:
@ezequielc yes, sorry. I had deployed Apps, but forgot to specify stage-3 (it was deployed to stage-0). It should be good now on stage-3 too.

ezequielc commented:
@fjvallarino Looks good. 👍
